stoppable_workers.go

```diff
@@ -68,6 +70,28 @@ func (sw *StoppableWorkers) Add(worker func(context.Context)) {
 	})
 }
 
+// StartTimer is an optional call that creates a timer that can be used in conjunction with
+// `NextTick`.
+func (sw *StoppableWorkers) StartTimer(duration time.Duration) {
+	sw.timer = time.NewTimer(duration)
+}
+
+// NextTick blocks until the timer ticks or the StoppableWorkers has been canceled. It returns true
+// if the timer has ticked and false if the context is canceled. Such that one can write a loop to
+// do work as such:
+//
+//	for sw.NextTick() {
```

Comment:
Hmm I like the ergonomics of this example. But:

You likely already alluded to this in our offline discussion, but my main concern is about multiple goroutines calling `NextTick()` concurrently. IIUC only one can `<-` a tick of the timer, so all concurrent `NextTick()` calls are subject to missing ticks `<-`ed by other goroutines. If you wanted multiple goroutines to be able to check ticks, you would need some "fan-out" pattern that would require more than one channel (or, hackily, relying on the fact that `close`ing a channel "notifies" all receivers).

Did you plan to limit usage of this to only one goroutine? If so, we'd probably need to document that and, preferably, somehow disallow calling `NextTick()` when another goroutine is already waiting for a tick. If you wanted more of "fan-out" pattern then would be happy to discuss alternate designs here.

Reply:
Talked offline -- I have no need for multiple goroutines being managed by a ticker (or set of tickers) event. @benjirewis and I decided to make a special constructor that would leverage the type system to prevent those usages.